### PR TITLE
ICU-20568 add unitPreferenceData to units.txt via ldml2icu rules.

### PR DESCRIPTION
**Pull request by hugovdm**
##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20568
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/data/misc/supplementalData.txt [hunk 1]
```diff
@@ -35596,385 +35596,6 @@ supplementalData:table(nofallback){
...
-                        unit{"square-mile"}
-                    }
-                }
-                US{
-                    {
-                        unit{"square-mile"}
-                    }
-                }
-            }
-            "land-agricult"{
```
Comment (hugovdm) [inline icu4c/source/data/misc/supplementalData.txt:35618]: CLDR simplified this to only "land", appearing in units.txt line 440.

File: tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt [hunk 2]
```diff
@@ -98,10 +98,32 @@
...
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/unit
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/geq ; values=$4
+//supplementalData/unitPreferenceData/unitPreferences[@category="(%W)"][@usage="(%A)"]/unitPreference[@regions="(%A)"][@skeleton="(%A)"]
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/unit
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/skeleton ; values=$4
+//supplementalData/unitPreferenceData/unitPreferences[@category="(%W)"][@usage="(%A)"]/unitPreference[@regions="(%A)"][@geq="(%A)"][@skeleton="(%A)"]
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/unit
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/geq ; values=$4
+    ; /unitPreferenceData/"$1"/"$2"/$3/<FIFO>/skeleton ; values=$5
+# No usage -> category default
```
Comment (hugovdm) [inline tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt]: CLDR added usage="default" stanzas, so I think I should drop these rules.

File: icu4c/source/data/misc/supplementalData.txt [hunk 3]
```diff
@@ -35596,385 +35596,6 @@ supplementalData:table(nofallback){
...
-                        unit{"square-meter"}
-                    }
-                    {
-                        unit{"square-centimeter"}
-                    }
-                }
-            }
-        }
-        "duration"{
-            "music-track"{
```
Comment (hugovdm) [inline icu4c/source/data/misc/supplementalData.txt:35687]: This became "media".

File: icu4c/source/data/misc/supplementalData.txt [hunk 4]
```diff
@@ -35596,385 +35596,6 @@ supplementalData:table(nofallback){
...
-                001{
-                    {
-                        unit{"year-person+month-person"}
-                    }
-                    {
-                        unit{"month-person"}
-                    }
-                }
-            }
-            "tv-program"{
```
Comment (hugovdm) [inline icu4c/source/data/misc/supplementalData.txt:35707]: This became "media".

File: icu4c/source/data/misc/supplementalData.txt [hunk 5]
```diff
@@ -35596,385 +35596,6 @@ supplementalData:table(nofallback){
...
-            _default{
-                001{
-                    {
-                        unit{"year"}
-                    }
-                    {
-                        unit{"month"}
-                    }
-                    {
-                        unit{"day"}
```
Comment (hugovdm) [inline icu4c/source/data/misc/supplementalData.txt:35726]: We're now starting with "day", not using "year" and "month".

File: icu4c/source/data/misc/units.txt [hunk 6]
```diff
@@ -374,4 +374,1408 @@ units:table(nofallback){
...
+                        unit{"yard"}
+                    }
+                    {
+                        unit{"yard"}
+                    }
+                }
+                SE{
+                    {
+                        geq{"0.1"}
+                        unit{"mile-scandinavian"}
```
Comment (hugovdm) [inline icu4c/source/data/misc/units.txt:1236]: We may need to add smaller units to SE.

Comment (hugovdm) [reply icu4c/source/data/misc/units.txt:1236]: This would be a CLDR change though, so considered out of scope for this pull request.

**Review by sffc — APPROVED** [review]
Very good, thanks!

There are some issues I noticed with the data, but those are CLDR bugs, out of scope of this PR.

**Pull request merged**